src/libraries/Microsoft.Extensions.Logging.EventSource/tests/EventSourceLoggerTest.cs

```diff
@@ -195,6 +195,67 @@ public void Logs_AsExpected_WithDefaults()
             }
         }
 
+        [Theory]
+        [InlineData(true, true, true)]
+        [InlineData(true, true, false)]
+        [InlineData(true, false, true)]
+        [InlineData(false, true, true)]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/73438", typeof(PlatformDetection), nameof(PlatformDetection.IsNativeAot))]
+        public void Logs_TracingDetailsAsExpected_WithDefaults(bool hasTrace, bool useW3CFormatId, bool sampled)
+        {
+            using (var testListener = new TestEventListener())
+            {
+                var factory = CreateLoggerFactory();
+
+                var listenerSettings = new TestEventListener.ListenerSettings();
+                listenerSettings.Keywords = (EventKeywords)(-1);
+                listenerSettings.FilterSpec = null;
+                listenerSettings.Level = default(EventLevel);
+                testListener.EnableEvents(listenerSettings);
+
+                using Activity activity = new Activity("TestOperation");
+
+                if (useW3CFormatId)
+                {
+                    activity.SetIdFormat(ActivityIdFormat.W3C);
+                }
+                else
+                {
+                    activity.SetIdFormat(ActivityIdFormat.Hierarchical);
+                }
```

Comment:
add these and `sampled` inside the `hasTrace` check?

Reply:
Updated